src/scanner/mod.rs

```diff
@@ -36,13 +40,20 @@ impl Scanner {
         ips: &[IpAddr],
         batch_size: u16,
         timeout: Duration,
+        tries: u8,
         greppable: bool,
         port_strategy: PortStrategy,
         accessible: bool,
     ) -> Self {
         Self {
             batch_size,
             timeout,
+            tries: if tries == 0 {
+                warning!("tries is 0, corrected to 1!");
+                NonZeroU8::new(1).unwrap()
+            } else {
+                NonZeroU8::new(tries).expect("tries cannot be 0!")
+            },
```

Comment:
Here we can do:

```rust
tries:  NonZeroU8::new(tries).unwrap_or(1)
```

I wouldn't bother adding a warn in this case. :)

Reply:
Almost, but the value in `unwrap_or` needs to be `NonZeroU8` too:
```rust
tries: NonZeroU8::new(tries).unwrap_or_else(|| NonZeroU8::new(1).unwrap()),
```

Reply:
I had trouble with this solution.
Indeed, when I tested it locally I got the following error messages from compiler:
`
tries: NonZeroU8::new(tries).unwrap_or(1),
   |                                                    ^ expected struct `std::num::NonZeroU8`, found integer
...
tries: NonZeroU8::new(tries).unwrap_or(1),
   |                                                    ^ expected struct `std::num::NonZeroU8`, found integer
`

I solved that by writing:
```rust
tries: NonZeroU8::new(tries).unwrap_or(NonZeroU8::new(1).unwrap()),
```
But I find this solution heavy... But maybe I am doing it wrong an there is a lighter solution.

Moreover I think it is important to inform the user that the code modified a parameter he/she gave on the CLI.

Reply:
It does not really get lighter than that. The shortest I could think of is

```rust
use std::cmp;
// ...
  tries: NonZeroU8::new(cmp::max(tries, 1)).unwrap(),
```

Reply:
Why not, but I think we need to discuss about informing or not the user?
What do you think about it?

Reply:
In my opinion I wouldn't worry about informing the user here, someone has to actively pass `0` to `tries`. It's more of a matter of protecting execution than user experience at this point.

Reply:
I am not really sure what's best here, so I am going to side with @bernardoamc. However, you could mention that `tries = 0` will be automatically changed to 1 in the manual page and maybe even in the help message.

Reply:
I just tested it and it runs fine, from commandline as well as from the config. The number gets changed to 1 if it's 0 and the warning pops up. And this is where I made a mistake by not explaining good enough how to use that macro. :/
We have an accessibility option, where people can chose to have the RustScan output in a screen-reader format, without ASCII art. Could you please change that macro by adding the 2 keyword that's already there in the context, so it gets printed with those options. 
```
warning!("tries is 0, corrected to 1!", greppable, accessible);
```

Reply:
I went with an `unwrap/cmp`-based solution and updated `README.md` as well as `--help` message.